### ISSUE TITLE
REST client - exception ODistributedOperationException - Incorrect quorum value .

### ISSUE THREAD

**Issue by letsbringachange** [orientechnologies/orientdb#8912]
## Expected behavior  
Index is successfully created on all the three nodes in the cluster.

## Actual behavior  
Index is created 2 of the 3 nodes and below exception received. default-distributed-db-config.json has writeQuorum as majority. We do not received any error message in server log files.

How is the quorum value being counted as 3, when we have set it to majority in a 3 node cluster.?

**Exception**
`com.orientechnologies.orient.server.distributed.task.ODistributedOperationException: **Quorum 3** not reached for request (id=2.81647 task=command_sql(create index `skill.id` on `skill` ( `id` ) UNIQUE_HASH_INDEX LONG)). Elapsed=13503ms. No server in conflict. Received: \\u000a - odsnode: 0\\u000a - nodemds1: 0\\u000d\\u000a\\u0009DB name=\\"DB1\\""\r\n    }\r\n  ]\r\n}`

Complete exception is attached.
[ods-client-exception.log](https://github.com/orientechnologies/orientdb/files/3303951/ods-client-exception.log)

## Steps to reproduce  
We are running batch inserts using the REST api to populate a DB with 300,000 plus vertices and indexes. Inserts are done in batches of 200 only.
Each time when job starts, it drops and re-creates the database. (We have limitations of doing delta imports, hence this current approach).
After 4-5 successful runs, the job doing the DB insert fails with the following error.

We are running a 3 node cluster with multicast disabled, and tcp-ip block containing the IP addresses of three nodes. Below is the JAVA process for OrientDB which shows all config parameters that we have used.

`/bin/java -Dstorage.useWAL=true -XX:+PerfDisableSharedMem -Dscript.pool.maxSize=2000 -Ddb.pool.max=2000 -Dstorage.wal.syncOnPageFlush=false -Dstorage.wal.allowDirectIO=false -Dstorage.diskCache.bufferSize=4096 -XX:MetaspaceSize=512m -XX:MaxMetaspaceSize=512m -XX:CompressedClassSpaceSize=512m -Dstorage.diskCache.writeCachePageTTL=3000 -server -Xms4G -Xmx4G -Djna.nosys=true -XX:+HeapDumpOnOutOfMemoryError -Djava.awt.headless=true -Dfile.encoding=UTF8 -Drhino.opt.level=9 -Ddistributed=true -Djava.util.logging.config.file=/opt2/orientdb/config/orientdb-server-log.properties -Dorientdb.config.file=/opt2/orientdb/config/orientdb-server-config.xml -Dorientdb.www.path=/opt2/orientdb/www -Dorientdb.build.number=3.0.x@rcc3c25ebc7d48a0db1073a0469cee2dd2947f4df; 2019-02-27 13:45:46+0000 -cp /opt2/orientdb/lib/orientdb-server-3.0.16.jar:/opt2/orientdb/lib/*:/opt2/orientdb/plugins/* com.orientechnologies.orient.server.OServerMain`

**Comment by letsbringachange** [comment]
Here are some more details from the logs after reducing the log level for distributed to FINE.


**Node 1**
`2019-06-19 18:46:59:370 FINE  [nodeodb3]<-[nodeodb1] Execute command=create index xyz.searchIndex on xyz ( _searchData ) FULLTEXT ENGINE LUCENE STRING db=DB1 [OSQLCommandTask]
2019-06-19 18:46:59:485 FINE  [nodeodb3]->[nodeodb1] Sending response 8683 back (reqId=id=1.19157 task=command_sql(create index xyz.searchIndex on xyz (_searchData ) FULLTEXT ENGINE LUCENE STRING)) [ODistributedWorker]
2019-06-19 18:46:59:490 FINE  [nodeodb3]<-[nodeodb1] Request id=1.19158 task=command_sql(drop index xyz.searchIndex) on database 'DB1' partitionKeys=[-2] task=command_sql(drop index award.searchIndex) [ODistributedDatabaseImpl]
`

**Node 2**
`2019-06-19 18:46:59:486 FINE  [nodeodb1]<-[nodeodb2] Received response '8683' for request (id=1.19157 task=command_sql(create index xyz.searchIndex on xyz ( _searchData ) FULLTEXT ENGINE LUCENE STRING)) (receivedCurrentNode=true receivedResponses=2 totalExpectedResponses=3 quorum=3) [ODistributedResponseManagerImpl]
2019-06-19 18:46:59:487 FINE  [nodeodb1]<-[nodeodb3] Received response '8683' for request (id=1.19157 task=command_sql(create index xyz.searchIndex on xyz ( _searchData ) FULLTEXT ENGINE LUCENE STRING)) (receivedCurrentNode=true receivedResponses=3 totalExpectedResponses=3 quorum=3) [ODistributedResponseManagerImpl]
2019-06-19 18:46:59:488 FINE  [nodeodb1]->[[nodeodb3, nodeodb2]] Distributed response (reqId=1.19157 quorum=3 result=null), checking for any fix needed... [ODistributedResponseManagerImpl]
2019-06-19 18:46:59:488 FINE  [nodeodb1] Sending signal of synchronous request completed (reqId=1.19157 thread=66) [ODistributedResponseManagerImpl]
2019-06-19 18:46:59:488 FINE  [nodeodb1] All synchronous responses collected in 118ms (reqId=1.19157 thread=12054) [ODistributedResponseManagerImpl]
2019-06-19 18:46:59:488 FINE  [nodeodb1] Detected 1 node(s) in timeout or in conflict and quorum (3) has not been reached, rolling back changes for request (id=1.19157 task=command_sql(create index xyz.searchIndex on xyz ( _searchData ) FULLTEXT ENGINE LUCENE STRING)) [ODistributedResponseManagerImpl]
2019-06-19 18:46:59:488 FINE  [nodeodb1] Quorum 3 not reached for request (id=1.19157 task=command_sql(create index xyz.searchIndex on xyz ( _searchData ) FULLTEXT ENGINE LUCENE STRING)). Elapsed=117ms. Servers in timeout/conflict are:
 - nodeodb1: 4342
Received:
 - nodeodb3: 8683
 - nodeodb2: 8683
 - nodeodb1: 4342 [ODistributedResponseManagerImpl]
2019-06-19 18:46:59:490 FINE  [nodeodb1]->[nodeodb3] Sending undo message (command_sql(drop index xyz.searchIndex)) for request (id=1.19157 task=command_sql(create index xyz.searchIndex on xyz ( _searchData ) FULLTEXT ENGINE LUCENE STRING)) database 'DB1' to server nodeodb3 [ODistributedResponseManagerImpl]
2019-06-19 18:46:59:490 FINE  [nodeodb1]->[[nodeodb3]] Sending request id=1.19158 task=command_sql(drop index xyz.searchIndex)... [ODistributedDatabaseImpl]`

**Comment by letsbringachange** [comment]
@lvca - By any chance would you be able to take a look at it ?

**Comment by letsbringachange** [comment]
@luigidellaquila - Would you be able to help with this?

**Comment by letsbringachange** [comment]
Tested today with 3.0.21-tp3 - Same error. Though the memory foot print on the node has reduced largely.

**Comment by letsbringachange** [comment]
Found the root cause to be delete of luceneIndexes folder when DB is dropped. A warning message is provided - unable to delete /opt2/orientdb/databases/DB1 since its is not empty Files [ opt2/orientdb/databases/DB1/luceneIndexes].

if the folder is deleted manually prior to create of DB again, LUCENE indexes are created successfully in the subsequent steps.

**Comment by jonsalvas** [comment]
Same issue here in 3.0.25 

Error: com.orientechnologies.orient.server.distributed.task.ODistributedOperationException: quorum of '3' not reached, responses: [node: pre-xxx-odb-orientdb-0 success,concurrent modification record (node pre-xxx-odb-orientdb-1): #119:2 database version: 66 transaction version: 65,concurrent modification record (node pre-xxx-odb-orientdb-2): #119:2 database version: 66 transaction version: 65]
How we reproduce the issue:
1. Make a lot of insertions (in best case with many conflicts) in parallel
2. We receive some "com.orientechnologies.orient.core.exception.OConcurrentModificationException: Cannot UPDATE the record #300:3 because the version is not the latest. Probably you are updating an old record or it has been modified by another user (db=v17 your=v16)" - which is ok so far
3. After some time we receive: Exception `708C0204` in storage `plocal:/orientdb/databases/xxx`: 3.0.25 - Veloce (build 2a229d542568b8d413b8633cca1f34239449d13e, branch UNKNOWN)
com.orientechnologies.orient.core.exception.OSerializationException: Found null entry in ridbag with rid=#244:-5
	DB name="xxx"
	at com.orientechnologies.orient.core.db.record.ridbag.embedded.OEmbeddedRidBag.serialize(OEmbeddedRidBag.java:440)
4. And finally: com.orientechnologies.orient.server.distributed.task.ODistributedOperationException: quorum of '3' not reached, responses: [node: pre-xxx-odb-orientdb-0 success,concurrent modification record (node pre-xxx-odb-orientdb-2): #119:2 database version: 66 transaction version: 65,concurrent modification record (node pre-xxx-odb-orientdb-1): #119:2 database version: 66 transaction version: 65]
5. As soon as it reaches this state: even if we stop the load and do one single insertion the error appears in log. The database seems to be completely blocked and cannot recover until we scale down and up the cluster.

Some side information: 
- We tried to write only to one node and let the two other nodes be readable. -> no improvement
- Initially we configured writeQuorum to majority. With three nodes, majority should be 2 but the error above says 3. Even if we hard set the value to 2 the error appears with 3, it seems to be ignored. 

@luigidellaquila could you PLEASE support here, we are loosing our ideas to fix the issue and our production release is on hold because of this. Let me know if you need any further info.
